specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,567 @@
+news article related notification is expected to launch the News app along with the article in question in
+the foreground. Another actionable scenario is for the user to actually interact with the contents
+of the notification popup like a UI control button. For example, a messaging app like Teams that displays a
+message from another user could have a "Respond To" textbox and button so that the user can directly
+respond to the message in the popup without having to launch the application. This scenario triggers
+a background process on behalf of the application (with no UI) which inturn processes the "Reply To"
+message and forwards the response to the other device.
+
+Notification messages could have a local or cloud based origin. In the case of local notifications, the message
+always originates from the app installed on the device. For cloud notifications, the message always
+originates from a remote application service that targets the locally installed app in question.
+
+Here is a visual representation of a simple notification with no interactive controls:
+
+![Notification Simple](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/images/send-toast-02.png)
+
+Here is a visual representation of a notification with simple button controls:
+
+![Notification Interactive 1](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/images/adaptivetoasts-structure.jpg)
+
+Here is a visual representation of a notification with a Message "Reply To" option:
+
+![Toast Interactive 2](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/images/toast-notification.png)
+
+For more details see:
+
+-   [Toast Notification WinRT APIs](https://docs.microsoft.com//uwp/api/Windows.UI.Notifications.ToastNotification?view=winrt-20348)
+    Defines all the API constructs that we have for notifications in WinRT today using the
+    Windows SDK.
+-   [Toast Notification UX Guidance using Toolkit](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/toast-ux-guidance)
+    Defines the UX Guidance for developers who want to display notifications using the toolkit libraries.
+-   [Sending Local Toasts using C# using Toolkit](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp)
+    Defines how an app developer can send interactive notifications to the user and setup activation
+    handlers for these notifications using the Toolkit libraries.
+
+## The problems today
+
+**Cloud Notifications integration with Windows App SDK**: While we support Push scenarios for Raw notification
+on behalf of unpackaged applications in the Windows App SDK 1.0, we do not support Cloud sourced App Notifications. We need
+to fill this gap for 1.1 and beyond.
+
+**API fragmentation**: There are too many API technologies that are published today to simply
```

Comment:
The goal here is that this covers all the cases for publicly distributed apps since this doc is geared towards the public WinAppSDK developer. 

Inbox components are a seperate discussion and can still continue using their choice of technology. Nothing stopping them from using reunion either if they need to.

Reply:
I had some time to process this question since I likely misunderstood it. Yes, the goal here is that the API super sets everything we have. From a notification perspective, it should cover every scenario that we already cover today + some more (as needed). As far as deprecation for inbox APIs are concerned, that would be a north star.

Reply:
But I understand WinAppSDK does not cover UWP's. Given that it is not a superset.

Reply:
Yes